Loop/Managers/LoopDataManager.swift

```diff
@@ -201,10 +201,11 @@ final class LoopDataManager {
         - recommendedTempBasal: The recommended temp basal based on predicted glucose
         - lastTempBasal:        The last set temp basal
         - lastLoopCompleted:    The last date at which a loop completed, from prediction to dose (if dosing is enabled)
+        - lastLoopError         The error, if any, from the last loop attempt
         - insulinOnBoard        Current insulin on board
         - error:                An error object explaining why the retrieval failed
      */
-    func getLoopStatus(resultsHandler: (predictedGlucose: [GlucoseValue]?, recommendedTempBasal: TempBasalRecommendation?, lastTempBasal: DoseEntry?, lastLoopCompleted: NSDate?, insulinOnBoard: InsulinValue?, error: ErrorType?) -> Void) {
+    func getLoopStatus(resultsHandler: (predictedGlucose: [GlucoseValue]?, recommendedTempBasal: TempBasalRecommendation?, lastTempBasal: DoseEntry?, lastLoopCompleted: NSDate?, lastLoopError: ErrorType?, insulinOnBoard: InsulinValue?, error: ErrorType?) -> Void) {
```

Comment:
so after looking more at this, it's pretty clear I haven't designed this API well. `getLoopStatus` actually triggers an `update` cycle, which exists because I wanted it to be possible for someone who opens the UI mid-loop to see the latest prediction data while avoiding doing any mid-loop calculations while the app isn't open.

I think your suggestion is best – `error ?? lastLoopError`. Because you're checking status immediately after looping, `error` should never be `nil` unless there's a data problem, in which case it's exactly the same as `lastLoopError`.